packages/back-end/src/controllers/admin.ts

```diff
@@ -96,3 +231,131 @@ export async function putOrganization(
     status: 200,
   });
 }
+export async function getMembers(
+  req: AuthRequest<never, never, { page?: string; search?: string }>,
+  res: Response
+) {
+  if (!req.superAdmin) {
+    return res.status(403).json({
+      status: 403,
+      message: "Only superAdmins can get all members",
+    });
+  }
+
+  const { page, search } = req.query;
+
+  const organizationInfo: Record<string, object> = {};
+  const allUsers = await getAllUsersFiltered(parseInt(page ?? "1"), search);
+  if (allUsers?.length > 0) {
+    for await (const user of allUsers) {
+      const orgs = await findOrganizationsByMemberId(user.id);
+      organizationInfo[user.id] = orgs.map((o) => ({
+        id: o.id,
+        name: o.name,
+        members: o.members.length,
+        role: o.members.find((m) => m.id === user.id)?.role,
+      }));
+    }
+  }
+
+  return res.status(200).json({
+    status: 200,
+    members: allUsers,
+    total: await getTotalNumUsers(search),
+    memberOrgs: organizationInfo,
+  });
+}
+
+export async function getOrganizationMembers(
+  req: AuthRequest<
+    null,
+    {
+      orgId: string;
+    }
+  >,
+  res: Response
+) {
+  if (!req.superAdmin) {
+    return res.status(403).json({
+      status: 403,
+      message: "Only superAdmins can get all members",
+    });
+  }
+  const { orgId } = req.params;
+
+  const org = await getOrganizationById(orgId);
+  if (!org) {
+    return res.status(404).json({
+      status: 404,
+      message: "Organization not found",
+    });
+  }
+
+  const members: UserInterface[] = [];
+  for await (const member of org.members) {
+    const user = await getUserById(member.id);
+    if (user) {
+      members.push(user);
+    }
+  }
+
+  return res.status(200).json({
+    status: 200,
+    members,
+  });
+}
+
+export async function putMember(
+  req: AuthRequest<{
+    userId: string;
+    name: string;
+    email: string;
+    verified: boolean;
+  }>,
+  res: Response
+) {
+  if (!req.superAdmin) {
+    return res.status(403).json({
+      status: 403,
+      message: "Only superAdmins can update members",
+    });
+  }
```

Comment:
We don't have to do it in this PR, but we could consider making an adminRouter and adding this as middleware to remove code duplication and make it less likely that someone forgets to add this check.